auth-web/src/components/auth/staff/DissolutionSchedule.vue

```diff
@@ -0,0 +1,296 @@
+                class="mr-3"
+                @click="triggerEditOnOff()"
+              >
+                Cancel
+              </v-btn>
+              <v-btn
+                color="primary"
+                large
+                @click="saveBtnClicked()"
+              >
+                Save
+              </v-btn>
+            </v-col>
+          </v-row>
+        </template>
+      </article>
+    </section>
+  </div>
+</template>
+
+<script lang="ts">
+import { Component, Emit, Vue } from 'vue-property-decorator'
+import { mapActions } from 'pinia'
+import { useStaffStore } from '@/stores/staff'
+
+@Component({
+  methods: {
+    ...mapActions(useStaffStore,
+      ['getDissolutionBatchSize',
+        'isDissolutionBatchOnHold',
+        'updateDissolutionBatchSize',
+        'updateDissolutionBatchOnHold'])
+  }
+})
+export default class DissolutionSchedule extends Vue {
+  $refs: {
+    numberOfBusinessesRef: any
+  }
+
+  // Local properties
+  menu = false
+  numberOfBusinesses = 0
+  isEdit = false
+  isOnHold = false
+
+  readonly getDissolutionBatchSize!: () => number
+  readonly isDissolutionBatchOnHold!: () => boolean
+  readonly updateDissolutionBatchSize!: (x: number) => void
+  readonly updateDissolutionBatchOnHold!: (x: boolean) => void
+
+  /**
+   * Set local properties to values from the store.
+   * TODO: Update once BE work is done.
+   */
+  mounted (): void {
+    this.numberOfBusinesses = this.getDissolutionBatchSize()
+    this.isOnHold = this.isDissolutionBatchOnHold()
+  }
+
+  /**
+   * Update whether the dissolution batch is paused or running.
+   * Emit the status to the parent to know whether the "Paused" badge is going to be shown.
+   * TODO: Modify/Update this once the BE is done. */
+  actionBtnClicked (): void {
+    this.isOnHold = !this.isDissolutionBatchOnHold()
+    this.onHoldUpdate(this.isOnHold)
+    this.updateDissolutionBatchOnHold(!this.isDissolutionBatchOnHold())
+  }
+
+  /**
+   * Save button is clicked. Update the dissolution batch size.
+   * Only save if the inputted number is valid.
+   * TODO: Implement logic (job) once the BE is done.
+   */
+  saveBtnClicked (): void {
+    if (this.$refs.numberOfBusinessesRef.validate()) {
+      this.updateDissolutionBatchSize(this.numberOfBusinesses)
+      this.triggerEditOnOff()
+    }
+  }
+
+  /** Edit or Cancel button is clicked. */
+  triggerEditOnOff (): void {
+    this.isEdit = !this.isEdit
+    // closing the menu
+    this.menu = false
+  }
+
+  /** The array of validations rule(s) for the Dissolution Batch Size text field. */
+  get dissolutionBatchSizeRules (): Array<(v) => boolean | string> {
+    return [
+      v => !!v || 'The number of businesses to be moved into D1 dissolution per batch. Maximum of 2000.',
+      v => (v % 1 === 0) || 'The number of businesses to be moved into D1 dissolution per batch. Maximum of 2000.',
+      v => v >= 0 || 'The number of businesses to be moved into D1 dissolution per batch. Maximum of 2000.',
```

Comment:
Did you mean to have the same 3 error messages?

The second check is for an even/odd value, right? The message doesn't seem to convey the rule.

Reply:
The Designs didn't show different error messages but showed them as errors. The second one is to check if the value has a decimal point.

Now that I'm thinking about, it's probably a good idea to bring it up to Andy to see if we're going to show different messages.

Reply:
Oh yeah. Checking for even/odd would be `% 2`.

You could also use `Number.isInteger()`. I'm OK with either.